src/Entity/User.php

```diff
@@ -0,0 +1,184 @@
+<?php
+
+namespace App\Entity;
+
+use App\Repository\UserRepository;
+use DateTimeImmutable;
+use Doctrine\ORM\Mapping as ORM;
+use Gedmo\Timestampable\Traits\TimestampableEntity;
+use Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity;
+use Symfony\Component\Security\Core\User\PasswordAuthenticatedUserInterface;
+use Symfony\Component\Validator\Constraints as Assert;
+
+#[ORM\Entity(repositoryClass: UserRepository::class)]
+#[ORM\Table(name: '`user`')]
+#[UniqueEntity(fields: ['email'], message: 'Email address should be unique')]
+#[UniqueEntity(fields: ['employeeId'], message: 'Employee ID should be unique')]
+class User implements PasswordAuthenticatedUserInterface
+{
+    use TimestampableEntity;
+
+    #[ORM\Id]
+    #[ORM\GeneratedValue]
+    #[ORM\Column]
+    private ?int $id = null;
+
+    #[ORM\Column(length: 255), Assert\NotBlank(message: 'Name should not be blank.')]
+    private ?string $name = null;
+
+    #[ORM\Column(length: 255)]
+    #[Assert\NotBlank(message: 'Email should not be blank.')]
+    #[Assert\Email(message: 'Please enter a valid email address')]
+    private ?string $email = null;
+
+    #[ORM\Column(length: 255)]
```

Comment:
The maximum length should not exceed 20 characters for mobile numbers. Also, we need to follow the other fields as needed.

Reply:
Done